Src/Support/GoogleApis.Auth.PlatformServices_Shared/OAuth2/Pkcs8.cs

```diff
@@ -0,0 +1,266 @@
+
+                public byte[] ReadBytes(int length)
+                {
+                    if (length <= 0)
+                    {
+                        throw new ArgumentOutOfRangeException(nameof(length), "length must be positive.");
+                    }
+                    if (_bs.Length - length < 0)
+                    {
+                        throw new ArgumentException("Cannot read past end of buffer.");
+                    }
+                    byte[] result = new byte[length];
+                    Array.Copy(_bs, _index, result, 0, length);
+                    _index += length;
+                    return result;
+                }
+
+                public Tag ReadTag()
+                {
+                    byte b = NextByte();
+                    int tag = b & 0x1f;
+                    if (tag == 0x1f)
+                    {
+                        throw new NotSupportedException("Tags of value > 30 not supported.");
+                    }
+                    else
+                    {
+                        return (Tag)tag;
+                    }
+                }
+
+                public int ReadLength()
+                {
+                    byte b0 = NextByte();
+                    if ((b0 & 0x80) == 0)
+                    {
+                        return b0;
+                    }
+                    else
+                    {
+                        if (b0 == 0xff)
+                        {
+                            throw new InvalidOperationException("Invalid length byte: 0xff");
+                        }
+                        int byteCount = b0 & 0x7f;
+                        if (byteCount == 0)
+                        {
+                            throw new NotSupportedException("Lengths in Indefinite Form not supported.");
+                        }
+                        int result = 0;
+                        for (int i = 0; i < byteCount; i++)
+                        {
+                            if ((result & 0xff800000) != 0)
+                            {
+                                throw new NotSupportedException("Lengths > 2^31 not supported.");
+                            }
+                            result = (result << 8) | NextByte();
+                        }
+                        return result;
+                    }
+                }
+
+                public int EndOffset(int length) => _index + length;
+
+            }
+
+            public static object Decode(byte[] bs) => new Decoder(bs).Decode();
+
+        }
+
+        public static RSAParameters DecodeRsaParameters(string pkcs8Base64)
+        {
+            const string PrivateKeyPrefix = "-----BEGIN PRIVATE KEY-----";
+            const string PrivateKeySuffix = "-----END PRIVATE KEY-----";
+
+            Utilities.ThrowIfNullOrEmpty(pkcs8Base64, "pkcs8PrivateKey");
+            var base64PrivateKey = pkcs8Base64.Replace(PrivateKeyPrefix, "").Replace(PrivateKeySuffix, "");
```

Comment:
Should we validate that it does actually start and end with this, then remove by length?